File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    phone_book.sort();
    for(let i =0;i<phone_book.length-1;i++){
        const prefixSize = phone_book[i].length;
        const prefix = phone_book[i+1].slice(0,prefixSize);
        if(phone_book[i]===prefix) return false;
    }
    return answer;
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    sort(phone_book.begin(),phone_book.end());
    for(int i = 0;i<phone_book.size()-1;i++){
        int stringSize = phone_book[i].size();
        string str = phone_book[i+1].substr(0,stringSize);
        if(phone_book[i] == str) return false;
    }
    return answer;
}